### PR TITLE
add doc for Custom LDAP Property Mapping

### PR DESCRIPTION
**Pull request by ChenWenBrian**
## Details

Add detailed steps for users to know how to add custom LDAP Property Mapping, cause this question is usually asked and got no answer.

---

## Checklist

-   [ ] Local tests pass (`ak test authentik/`)
-   [ ] The code has been formatted (`make lint-fix`)

If an API change has been made

-   [ ] The API schema has been updated (`make gen-build`)

If changes to the frontend have been made

-   [ ] The code has been formatted (`make web`)

If applicable

-   [x] The documentation has been updated
-   [x] The documentation has been formatted (`make website`)

### CODE REVIEW

**Comment by tanberry** [comment]
Thanks very very much for this contribution, @ChenWenBrian ! We definitely need more procedural docs (the How To, with steps, kind of docs), so this is a great addition.

I'll make some edits (grammar and style), then after you make those changes please be sure to run either `make website` or `npm run prettier` before you push the chanes. The build is currently failing on `lint_prettier`.

Thanks again!

File: website/docs/property-mappings/index.md [hunk 1]
```diff
@@ -23,6 +23,20 @@ LDAP Property Mappings are used when you define a LDAP Source. These mappings de
...
+
+If the default source mapping is not enough, you can set your own custom property mapping. 
+
+For example `ldap-displayName-mapping:name`, which means ldap source field `displayName` will be mapped to `name` field in authentik. Here's the steps:
+
+1. Open admin UI, navigate to `Customization` -> `Property Mappings`.
+2. Click `Create`, select `LDAP Property Mapping` and then click `Next`.
+3. Type a unique and meaningful `Name`, such as `ldap-displayName-mapping:name`
+4. Type authentik inner field in `Object field`. such as `name`. If you want to add more extend attributes, you can type `attributes.mobile` for example.
+5. Type `Expression` which will get value from LDAP source. e.g. `return list_flatten(ldap.get("displayName"))`.
```
Comment (tanberry) [inline website/docs/property-mappings/index.md:36]:
```suggestion
5. In the **Expression**  field enter Pythin expressions to retrieve the value from LDAP source. For example `return list_flatten(ldap.get("displayName"))`.
```

File: website/docs/property-mappings/index.md [hunk 2]
```diff
@@ -23,6 +23,20 @@ LDAP Property Mappings are used when you define a LDAP Source. These mappings de
...
+
+`list_flatten(["input string array"])` will convert string array to single string. If you are not sure whether the ldap field is an array or not, you may map the field to any `attributes.xxx` and then check the sync result in authentik UI.
```
Comment (tanberry) [inline website/docs/property-mappings/index.md:38]:
```suggestion
`list_flatten(["input string array"])` will convert a string array to a single string. If you are not sure whether the LDAP field is an array or not, you can map the field to any `attributes.xxx` and then check the sync result in authentik UI.
```

**Review by tanberry — CHANGES_REQUESTED** [review]
Thanks so much @ChenWenBrian for these steps, it is great to have more procedural/How To docs.

I made quite a few copy-edits, but I made then as Suggestions so that you can easily accept them. Let me know if you have any questions.

Also, please be sure to run `make website` or `npm run prettier` before you push your changes, so that the build checks pass.

**Comment by tanberry** [comment]
Hi again, thanks for the speedy fixes. After checking on a few technical details with @BeryJu we think this content might actually better belong on our documentation about LDAP sources, but I want to look at it again tomorrow and get back to you. Wherever it ends up, we very much appreciate having it added to the docs.

**Comment by ChenWenBrian** [comment]
Hi @tanberry , thanks for your help. I accepted all your changes and fixed a few typos.

I haven't prepare a dev environment for this project, just update the doc from github UI. so, hope this won't bring too much trouble🙏

**Comment by tanberry** [comment]
Hi again @ChenWenBrian apologies for my delay. We do want to move this content to a different section in the docs; we think it would fit better here: https://docs.goauthentik.io/docs/sources/ldap/#property-mappings

Let me know if you want to do this yourself (either in this PR or in a new one). Or if not, I can move it. Either way, thank you for adding this steps!

**Comment by ChenWenBrian** [comment]
@tanberry please help to move the content, cause i'm not familiar with the procedure, thanks

**Comment by tanberry** [comment]
Closing this one since the work is now done in https://github.com/goauthentik/authentik/pull/10158. Thanks @ChenWenBrian!

**Pull request closed without merging**